torch/csrc/jit/codegen/cuda/dynamic_type.h

```diff
@@ -0,0 +1,299 @@
+#pragma once
+
+#include <c10/macros/Export.h>
+#include <c10/util/Exception.h>
+#include <cmath>
+#include <iostream>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+class TORCH_CUDA_CU_API IntOrDouble {
```

Comment:
nit: can this be simplified with `c10::variant`?

Reply:
WOW, I was looking at `std::variant` and unfortunately, it was C++17 only. Didn't know that there is a `c10::variant`. Thanks for pointing this out.